configuration/scripts/ratos-logging.sh

```
@@ -0,0 +1,238 @@
+    if [[ -n "$context" ]]; then
+        context_json=",\"context\":\"$context\""
+    fi
+    
+    # Build error code JSON
+    local error_code_json=""
+    if [[ -n "$error_code" ]]; then
+        error_code_json=",\"errorCode\":\"$error_code\""
+    fi
+    
+    # Create log entry
+    local log_entry="{\"level\":$level_value,\"time\":\"$timestamp\",\"msg\":\"$escaped_message\",\"source\":\"ratos-update\"$context_json$error_code_json,$(echo "$process_info" | sed 's/^{//;s/}$//')}"
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)